packages/snap-preact-components/src/components/Trackers/Recommendation/ResultTracker/RecommendationResultTracker.test.tsx

```diff
@@ -182,6 +188,240 @@ describe('RecommendationResultTracker Component', () => {
+		const controller = new RecommendationController(recommendConfig, {
+			client: new MockClient(globals, {}),
+			store: new RecommendationStore(recommendConfig, services),
+			urlManager,
+			eventManager: new EventManager(),
+			profiler: new Profiler(),
+			logger: new Logger(),
+			tracker: new Tracker(globals, { mode: 'development' }),
+		});
+
+		const trackfn = jest.spyOn(controller.tracker.track, 'event');
+
+		await controller.search();
+
+		const rendered = render(
+			<RecommendationProfileTracker controller={controller}>
+				<div>
+					{controller.store.results.map((result, idx) => (
+						<RecommendationResultTracker controller={controller} result={result} track={{ click: false }}>
+							<div className={'findMe'} key={idx}>
+								<div className="result">{result.mappings.core?.name}</div>
+							</div>
+						</RecommendationResultTracker>
+					))}
+				</div>
+			</RecommendationProfileTracker>
+		);
+
+		await waitFor(() => {
+			expect(trackfn).toHaveBeenCalledTimes(21);
+		});
+
+		// other 20 calls are for product render
+		controller.store.results.map((result) => {
+			expect(trackfn).toHaveBeenCalledWith({
+				type: BeaconType.PROFILE_PRODUCT_RENDER,
+				category: BeaconCategory.RECOMMENDATIONS,
+				context: controller.config.globals.siteId ? { website: { trackingCode: controller.config.globals.siteId } } : undefined,
+				pid: controller.events.render!.id,
+				event: {
+					context: {
+						placement: controller.store.profile.placement,
+						tag: controller.store.profile.tag,
+						type: 'product-recommendation',
+					},
+					product: {
+						id: result.id,
+						seed: undefined,
+						mappings: {
+							core: result.mappings.core,
+						},
+					},
+				},
+			});
+		});
+
+		trackfn.mockClear();
+
+		for (let i = 0; i < 21; i++) {
+			// @ts-ignore
+			const [callback] = window.IntersectionObserver.mock.calls[i];
+
+			callback([
+				{
+					isIntersecting: true,
+					intersectionRatio: 10,
+				},
+			]);
+		}
+
+		await waitFor(() => {
+			expect(trackfn).toHaveBeenCalledTimes(21);
+		});
+
+		//results should have done impression tracking
+		controller.store.results.map((result) => {
+			expect(trackfn).toHaveBeenCalledWith({
+				type: BeaconType.PROFILE_PRODUCT_IMPRESSION,
+				category: BeaconCategory.RECOMMENDATIONS,
+				context: controller.config.globals.siteId ? { website: { trackingCode: controller.config.globals.siteId } } : undefined,
+				pid: controller.events.impression?.id,
+				event: {
+					context: {
+						placement: controller.store.profile.placement,
+						tag: controller.store.profile.tag,
+						type: 'product-recommendation',
+					},
+					product: {
+						id: result.id,
+						seed: undefined,
+						mappings: {
+							core: result.mappings.core,
+						},
+					},
+				},
+			});
+		});
+
+		trackfn.mockClear();
+
+		const resultElem = rendered.container.querySelector('.findMe');
+
+		await userEvent.click(resultElem!);
+
+		expect(trackfn).toHaveBeenCalledWith({
```

Comment:
It seems the test is verifying that the profile click event has happened, but how can we be sure that the `trackfn` _has not_ been called with a product click event?